packages/indexer-generator/src/render-domain.ts

```diff
@@ -48,18 +49,45 @@ export class AccountDomain {
     return this.timeSeriesStats.getStats()
   }
 
-  async getEventsByTime(
-    startDate: number,
-    endDate: number,
-    opts: any,
-  ): Promise<StorageStream<string, ${Name}Event>> {
-    return await this.eventDAL
+  async getEvents(
+    args: EventsFilters
+  ): Promise<${Name}Event[]> {
+    const {
+      types,
+      startDate = 0,
+      endDate = Date.now(),
+      limit = 1000,
+      skip = 0,
+      reverse = true,
+    } = args
+
+    if (limit < 1 || limit > 1000)
+      throw new Error('400 Bad Request: 1 <= limit <= 1000')
+
+    const typesMap = types ? new Set(types) : undefined
+    const from = startDate ? [this.info.address, startDate] : [this.info.address]
+    const to = endDate ? [this.info.address, endDate] : [this.info.address]
+    let sk = skip
+
+    const result: ${Name}Event[] = []
+    const events = await this.eventDAL
       .useIndex(EventDALIndex.AccountTimestamp)
-      .getAllFromTo(
-        [this.info.address, startDate],
-        [this.info.address, endDate],
-        opts,
-      )
+      .getAllFromTo(from, to, { reverse, limit })
```

Comment:
Is this not causing issues when selecting the `EventDALIndex.AccountTimestamp` index, which has two keys, but then only passing in one key?

Reply:
there are two keys, from and to, maybe add default values for startDate (0) and endDate (Date.now())

Reply:
On the other hand, this may be disadvantageous in performance, compared to just using the Account index (if there is one).

Reply:
But yeah, given that both `endDate` and `startDate` are optional, this could actually be the way forward.